docs/contributing/contributing-code/contributing-code-control-plane/running-controlplane-locally.md

```diff
@@ -105,17 +96,73 @@ Run this command to create the namespace that will be used to store data.
 kubectl create namespace radius-testing
 ```
 
-## Setup Step 3: Create Resource Group and Environment
+## Setup Step 4: Setup Deployment Engine 
+
+
+> 💡 The Bicep deployment engine uses .NET. However you don't need to know C# or .NET to develop locally with Radius.
+> 💡 This way of setting up deployment-engine is useful if you are an external contributor and do not have access to `radius-project/deployment-engine` repo.
+> 💡 If you have access to the deployment-engine repository and would like to debug it, you can omit this step and proceed to step 5.
+
+### Setup Docker
+
+If Docker is not already installed, 
+* Download and install it from the [Docker Desktop download page](https://www.docker.com/products/docker-desktop). 
+Choose the installer that matches your operating system.
+* Open a terminal and run the following command to verify that Docker is installed and running:
+```sh
+docker --version
+```
+You should see the Docker version information.
+
+### Run Deployment Engine as a Docker container
+
+Run the below command.
+
+```sh
+docker run  -e RADIUSBACKENDURL=http://host.docker.internal:9000/apis/api.ucp.dev/v1alpha3 -p 5017:8080 ghcr.io/radius-project/deployment-engine:latest
+```
+
+`host.docker.internal` is a special DNS name provided by Docker that allows containers to access services running on the host machine 
+
+### Update launch.json 
+
+Open launch.json and comment out `Launch Deployment Engine` in `Launch Control Plane (all)`. The debug setup will use the Deployment Engine running as docker container. 
+  ```json
+  "compounds": [
+    {
+      "name": "Launch Control Plane (all)",
+      "configurations": [
+        "Launch UCP",
+        "Launch Applications RP",
+        "Launch Dynamic RP",
+        "Launch Controller",
+        // "Launch Deployment Engine"
+      ],
+      "stopAll": true
+    }
+  ],
+  ```
+
+## Setup Step 5 (optional): Setup deployment Engine for debugging
+
+If you have access to `radius-project/deployment-engine` repo, you can follow the steps below to set up Deployment Engine for debugging.
```

Comment:
```suggestion
If you have access to `radius-project/deployment-engine` repo, you can follow the steps below to set up Deployment Engine for debugging instead of running it in a docker container.
```